src/routes/v1/tokens.go

```diff
@@ -0,0 +1,86 @@
+package v1
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/dana-team/platform-backend/src/customerrors"
+	"github.com/dana-team/platform-backend/src/types"
+
+	"github.com/dana-team/platform-backend/src/controllers"
+	"github.com/dana-team/platform-backend/src/middleware"
+	"github.com/gin-gonic/gin"
+)
+
+// tokenHandler wraps a handler function with context setup for tokenController.
+func tokenHandler(handler func(controller controllers.TokenController, c *gin.Context) (interface{}, error)) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		kubeClient, err := middleware.GetKubeClient(c)
+		if middleware.AddErrorToContext(c, err) {
+			return
+		}
+
+		logger, err := middleware.GetLogger(c)
+		if middleware.AddErrorToContext(c, err) {
+			return
+		}
+
+		context := c.Request.Context()
+		tokenController := controllers.NewTokenController(kubeClient, context, logger)
+
+		result, err := handler(tokenController, c)
+		if middleware.AddErrorToContext(c, err) {
+			return
+		}
+
+		c.JSON(http.StatusOK, result)
+	}
+}
+
+// CreateToken returns a Gin handler function for creating a service account token.
+func CreateToken() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var request types.ServiceAccountRequestUri
+		if err := c.BindUri(&request); err != nil {
+			middleware.AddErrorToContext(c, customerrors.NewValidationError(err.Error()))
+			return
+		}
+
+		var query types.CreateTokenQuery
+		if err := c.BindQuery(&query); err != nil {
+			middleware.AddErrorToContext(c, customerrors.NewValidationError(err.Error()))
+			return
+		}
+		expirationSeconds := query.ExpirationSeconds
+		if query.ExpirationSeconds == "" {
+			expirationSeconds = "36000"
+		}
+		expireIn, err := strconv.Atoi(expirationSeconds)
```

Comment:
I don't think this piece of logic belongs here, should be in the controllers layer instead